### PR TITLE
Omniwallet Rebrand

### PR DESCRIPTION
**Pull request by Nevtep**
Long list of changes, full rebrand of omniwallet designed by warren and developed by me, including several new pages, i18n, added functionality.

Can be seen at staging.omniwallet.org

Testing is appreciated please report any quirks here and I will fix ASAP

### CODE REVIEW

**Comment by achamely** [comment]
@Nevtep   looks awesome so far. 
My observations on staging:
- Initial load of the overview page throws a bunch of console errors . These do disappear after the overview is loaded
    (TypeError: Cannot read property 'length' of undefined) and (Error: Invalid value for <g> attribute transform="translate(NaN,NaN)")
-- Navigating away then back causes console errors (TypeError: Cannot read property 'divisible' of undefined, at self.getPendingPos (AddressModelFactory.js:80))
--- Doesn't look like it affects display but still something to look into. 
- Trade page: the BTC/TMSC button changes the heading/titles but the actual transaction list/information remains unchanged. 
- Some copy changes and site links we'll need to do
-- MSC is now OMNI, Mastercoin Foundation is now Omni Foundation

**Comment by Nevtep** [comment]
Ok, changing msc to omni will break the app untill the db updates, the rest is fixed and up on staging

**Comment by dexX7** [comment]
1) One of my balances is not correct, likely due to number precision. Instead of `9223372036854775807` tokens a balance of `9223372036854775808` is shown on a) overview, b) my addresses, c) send dialog. d) in the history two different amounts are shown, whereby none of them is correct. e) It's also an issue on the transaction inspector. The amount is correctly shown under "my assets". I haven't checked other numbers.

maxint

maxint_addresses

maxint_send

maxint_history

maxint_inspector

Please note that Omni amounts use all 64 bits.

2) If it's possible and easy to do, asset names should be truncated on the a) overview, b) my addresses, c) my assets, d) send dialog, e) history, otherwise it can heavily overflow. f) This appears also to be an issue on "explorer - assets".

File: bower.json [hunk 1]
```diff
@@ -5,24 +5,31 @@
...
     "forge": "https://github.com/digitalbazaar/forge.git#0.2.30",
+    "angular-password-strength": "https://github.com/curtislacy/angular-password-strength.git#Caret-Operator",
+    "angular-timer": "~1.1.6",
+    "ngInfiniteScroll": "1.0.0",
     "numeraljs": "https://github.com/adamwdraper/Numeral-js.git#1.5.3",
     "angular-numeraljs": "https://github.com/sagyeli/angular-numeraljs.git#master",
-    "angular-password-strength": "https://github.com/curtislacy/angular-password-strength.git#Caret-Operator",
-    "angular-idle": "https://github.com/hackedbychinese/ng-idle.git#0.3.5",
+    "ng-idle": "https://github.com/hackedbychinese/ng-idle.git#0.3.5",
     "angular-re-captcha": "~0.0.3",
```
Comment (achamely) [inline bower.json]: @Nevtep   i've forged all these individual user repos to the omniwallet repo. 
Can we please go ahead an update all the direct github links to use the projects as they are forked on https://github.com/Omniwallet

**Comment by achamely** [comment]
@Nevtep   
* homepage balance checker is missing the exodus address 1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P  in the example
* under the my assets page clicking on an asset name / going to the asset detail page is resulting in all variable name placeholders with no data
image

* History page tx's don't appear to be ordered by time properly

**Comment by Nevtep** [comment]
3) It seems the send is not functioning well indeed but this is more related to the backend than the rebrand itself.

1) this is what I see in the transaction explorer, let me know if it is correct and if you see the same data on the page 
image

**Comment by Nevtep** [comment]
@dexX7  I've patched a place on the balance that was missing the big.js wraping to handle 64bit, let me know if you still see anything wrongly displayed.

@achamely send functionality hasn't changed from master, can you confirm that the send errors dex reports are not happening on main branch? maube something changed in prod environment? I seem to be able to send coins without problem *most of the time* but sometimes errors appear on production as well as on staging. can we track this as a different bug please?

I fixed some things that didn't seem to be working but now it should. Staging needs to pull from rebrand branch I can't access now ( on a different ip) I will merge changes tomorrow afternoon, can we agree on what's left to merge this in and track the send bug separately? as I don't think it is related to this PR

**Comment by dexX7** [comment]
@Nevtep: thanks for tackling this! 

It is fixed in some places, but there are still some missing:

1) Overview:

1_overview

2) My addresses:

2_myaddresses

3) My addresses:

3_myaddresses

4) History:

4_history

5) Send dialog:

5_send

**Comment by achamely** [comment]
@Nevtep we pushed a patch to production that involved using the updated bitcoinjs-lib repo hosted on omniwallet.co  github (@dexx7 its the low s-value issue)

**Comment by patrickdugan** [comment]
I am forced to place whole integer amounts of OMNI/MSC to engage an order, but the actual amount of OMNI that is tranascted is the Willet quantity. Thus I think we are in a situation where the UI option that was removed has not been fully reconciled. Please set is 1 OMNI = 1 OMNI in the order not .00000001


Also the miner fee of .0001 is prone to very llong delays in getting a confirmation, we need to set it higher or go with a Bitcoin Core RPC to set it.

**Comment by achamely** [comment]
@patrickdugan can you provide a screenshot (or details) of exactly what values you entered into the form. 
Also the miners fee is adjustable (but yes, dexx already brought up the point of making it more dynamic which we will be looking into)

**Comment by dexX7** [comment]
According to Omni Wallet, I own 55,011,739,443,812,007,936.00 $ worth of BTC. ;)

Also: new truncation issues:

rich

**Comment by Nevtep** [comment]
Alright, after a hard work yesterday I finished fixing the balances, there where some probles due to the backend returning strings for numbers now (since 64bit ints will get truncated by javascript natively) and also some rounding issues when using float in python that also got fixed. I tested the exchange and it looks like it works.

Please let me know if anything else is left for wrapping this up. I understand we need to change the way we address miner's fees right now, but I preffer that be treated as a different issue.

Thanks

**Comment by Nevtep** [comment]
Found and fixed the willet bug on buy order page, also addresses wheren't showing on the dropdown, fixed as well

**Comment by Nevtep** [comment]
Are we ready to merge this and move forward?

**Comment by achamely** [comment]
@Nevtep found a small bug 
https://staging.omniwallet.org/assets/details/2147483721
some variables aren't being displayed same issue for the history tab

**Comment by Nevtep** [comment]
Fixed

**Comment by dexX7** [comment]
It looks like my issues got fixed. Thanks! No merge objections from me.

**Comment by achamely** [comment]
Cool, That looks good now..  Though i am noticing 2 more things. 
1. After doing a send 2 things happen that are unexpected. 
  a- The Green confirmation modal is not dismissable (no 'x' to close it)
  b - After a send i'm redirected to the overview page (though the menu bar showed the 'My Addresses' highlighted. 
2. I just completed a send and I have no UI updates/feedback for the pending balance changes. 
    I had to log out/ log back in for the pending balance displays to be shown.

**Comment by achamely** [comment]
I'll aim to merge/deploy first thing tomorrow if everything checks out

**Comment by dexX7** [comment]
Did @zathras-crypto give his feedback? If I recall correctly, he mentioned something about testing OW and he prepared a list with items.

**Comment by zathras-crypto** [comment]
Sorry guys...

So I tested staging a bit, and found the following issues:

* Enter key doesn't work on password field during login
* Wallets from production are not accessible in staging
* On overview list of assets, assets not vertically centered in their shaded bars
* Refreshing page seems to log you out sometimes
* Values for DEx trades are orders of magnitude off (eg showing 2 for 20 etc)

Other things that are not necessarily bugs:

* No visibility of inbound unconfirmed transactions (Omni Core can provide this data FYI)
* Transaction cost seems to imply there is also an Omni transaction fee which is about 3 times higher than Bitcoin fee.  Perhaps it could be reworded, but most importantly perhaps switch to Class C where you don't need to spend 25,000 satoshi on outputs.

**Comment by achamely** [comment]
@zathras-crypto   
- Staging and production use separate databases and separate wallets .  After the rebrand i'll be merging the 2 databases to unify the system
- Refreshing via F5 , refresh button will log you out right now. 
- As soon as we merge the rebrand we'll be Replacing all the Mastercoin names with omni. 
- Incoming unconfirmed only exist if the tx was created within omniwallet,  once the rebrand is merged we'll have quite a bit of updates to start incorporating more data directly from core (including fee's and class C tx creations)

Remember, right now this is just a face lift for the UI, most of the underlaying logic comes next

**Comment by achamely** [comment]
@Nevtep   found 2 more bugs.

On the overview graph the 'Estimated Total value' is blank/empty
The portfolio Composition by value asset list includes random entries for coins it think are named 0 or 1?
overview graph

**Comment by achamely** [comment]
@Nevtep 
If any of these remaining issues are easy fixes let's get them in.  If they are going to require an unusually long amount of effort I'm ok with merging and then we can start opening issues to address those.  Just let me know

**Comment by marvgmail** [comment]
When Account Settings is opened, this error message is displayed. It's repeatable and happens to others.
image

**Comment by marvgmail** [comment]
Two things about https://staging.omniwallet.org/exchange/myoffers

1. There's no mention of the properties involved

1. Even though I have Display Test Ecosystem turned off in preferences, this page shows me a Test ecosystem offer I have pending.

**Comment by marvgmail** [comment]
On History page:

1. Test ecosystem history displayed even though it's turned off in my preferences

1. "Full Wallet History" doesn't include BTC transactions, so it's not "Full". How about "Omni Layer Wallet History"?

1. Most transactions don't display a transaction time. Transaction Inspector doesn't have a time for these transactions either.

1. "You put -3 Test MasterCoin for sale" - the amount in the tx details field should be a positive number.

1. Widen field or shrink font so transaction time value doesn't wrap.

1. font size problem 
image

**Comment by marvgmail** [comment]
Notice the date on https://staging.omniwallet.org/assets/details/2147483746: 
image

**Comment by marvgmail** [comment]
Transaction inspector doesn't have the currency name for properties other than MSC. For instance, https://staging.omniwallet.org/explorer/inspector?view=10d8b4cf8e84e9ef30395f5743798cc2c71a7bb1814b25f1cc87e357cd15b3d4
image

**Comment by zathras-crypto** [comment]
> Remember, right now this is just a face lift for the UI, most of the underlying logic comes next

Thanks bud - FYI I think I must have been testing at the same time as Tomas was working because I retested some of the things I originally found and they'd been fixed (like wallet history not appearing and such)

Looking good :)

**Comment by achamely** [comment]
reports from @patrickdugan that the accept offer page button is not creating transactions which include the omni exodus marker. Might be a simple matter of making sure we are passing the marker=true option to the api call to create transaction

**Comment by Nevtep** [comment]
Adam, the only thing I notice is some TX in history comming as type -1, and I don't have translation for those, not sure what those are. Other than that I see this is closed and we can address further changes as separate requirements. can we merge this and move on to renaiming msc/omni mastercoin/omni foundation?

thanks

**Comment by achamely** [comment]
@Nevtep txtype -1 are unknown/invalid Omni transactions but valid BTC transactions.  We keep them around as placeholders of attempted transactions so we can display to the user whatever 'attempt' they made even if it was invalid. I.e. failed feedback is better than no feedback

**Comment by Nevtep** [comment]
Adam, bugs have been fixed but I'm seeing 502 errors in the explorer, not sure why since I didn't made changes there, I'm looking into it

**Comment by Nevtep** [comment]
Really good progress today, made a whole flow of the dex and got it right and working, please let me know if there's anything else left to merge this and move forward :)

**Comment by achamely** [comment]
@Nevtep i've reviewed everything patched so far and it does indeed look good. 
I see 1 bug we need to still address:   logout

And 2 wording changes/tweaks:
-On the asset detail page we have the tx history labeled as Crowdsale History. As it stands not all assets are crowdsales. So we should change the Hedaing from " Crowdsale History" ->  "Asset History"
-On the history page lets change the wording "Full History" -> "Omni Protocol History"

**Comment by achamely** [comment]
@Nevtep Took a look at the logout a little more.  Part of the issue of the 'extra' menu sticking around is because how we do ng-if
ex: submenu is rendered when  <div id="wallet-navbar" ng-if="section=='wallet'">  etc...  

When we trigger the logout for the user we don't remove the 'section' variable
if we add / change the ng-click for logout to   ng-click="logout(); section='';"  it properly clears the menu, 

However there still appears to be a bug with logging in at that point that i'm tracking down

**Comment by achamely** [comment]
@Nevtep   added some logging and found the following was being generated by the login controller when a user tried to login after having logged out.  Possibly all we need to do i clear/disconnect the balance socket for the user?


TypeError: Cannot read property 'on' of null
    at BalanceSocketService.self.on (https://staging.omniwallet.org/js/services/BalanceSocket.js:16:15)
    at self.initialize (https://staging.omniwallet.org/js/factories/AddressModelFactory.js:24:20)
    at new AddressModelFactory.AddressModel (https://staging.omniwallet.org/js/factories/AddressModelFactory.js:94:10)
    at WalletService.self._addAddress (https://staging.omniwallet.org/js/services/WalletService.js:76:25)
    at https://staging.omniwallet.org/js/services/WalletService.js:24:23
    at Array.forEach (native)
    at WalletService.self.initialize (https://staging.omniwallet.org/js/services/WalletService.js:23:31)
    at self.verify.then.self.loginInProgress (https://staging.omniwallet.org/js/services/AccountService.js:151:36)
    at wrappedCallback (https://staging.omniwallet.org/bower_components/angular/angular.js:11722:81)
    at wrappedCallback (https://staging.omniwallet.org/bower_components/angular/angular.js:11722:81)

**Comment by Nevtep** [comment]
It's actually a bit more delicate because wallet and account services need to be disposed... I'll look into it and add the other changes now

**Comment by Nevtep** [comment]
@achamely logout is fixed as well as wordings, balancesocket was disconnecting but .connected variable inside needed to be set to false for addresses to work properly

all done sir, let me know what you think

**Comment by achamely** [comment]
@Nevtep initial tests looking very good. I'll give it some more review monday and then lets aim for merge tuesday morning :)

**Comment by achamely** [comment]
@Nevtep Found a really weird one with Marv this morning when finishing testing.
The logout bug was fixed however it has revealed a secondary bug/issue. 

Upon first login my address lists is populated just fine and assets are displayed properly.
If i then logout and log back in (without refreshing/reloading the tab/page) the overview and my address (viewed by asset) pages show my address doubled/listed twice. If i logout / log back in again it triples the display.  Additionally the chart does not load on subsequent logins until you change tabs and then come back to overview page.

**Comment by achamely** [comment]
Additionally the history page seems to not display full data and when you click the inspector for one of those tx's it takes you to a weird currency display (Though searching for that txid in the normal explorer works and displays correctly)

**Comment by achamely** [comment]
closing this pr as it's been merged to it's own branch (rebrand-v2) and a new pr https://github.com/OmniLayer/omniwallet/pull/1264  opened to track the final merge

**Pull request closed without merging**